app/services/users/remove_role.rb

```diff
@@ -0,0 +1,49 @@
+module Users
+  class RemoveRole
+    attr_reader :user, :role, :resource_type, :current_user, :response
+
+    Response = Struct.new(:success, :error_message, keyword_init: true)
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def initialize(user:, role:, resource_type:, current_user:)
+      @user = user
+      @role = role
+      @resource_type = resource_type&.safe_constantize
+      @current_user = current_user
+      @response = Response.new(success: false)
+    end
+
+    def call
+      return response if super_admin_role?(role)
+      return response if user_is_current_user?(user)
+
+      if !resource_type.nil? && @user.remove_role(role, resource_type)
```

Comment:
Right, this is probably a really bad idea in terms of readability, but if you wanted to simplify this ` if-elsif-else` , this would work:

```ruby
if @user.remove_role(*[role, resource_type].compact)
```

Reply:
This is such a neat way to refactor this! I decided to refactor this line to the following for readability purposes, but TIL:
```
 if resource_type.present? && @user.remove_role(role, resource_type)
```